### PR TITLE
highlight pillar looking at and add removal of a pillarstep

### PR DESCRIPTION
**Pull request by NilsAffing**

### CODE REVIEW

File: client/shader/plant_dummy.tcs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (ID == 0) {
+        gl_TessLevelInner[0] = tess_level_inner;
+        gl_TessLevelInner[1] = tess_level_inner;
+        gl_TessLevelOuter[0] = tess_level_outer;
+        gl_TessLevelOuter[1] = tess_level_outer;
+        gl_TessLevelOuter[2] = tess_level_outer;
+        gl_TessLevelOuter[3] = tess_level_outer;
+    }
+}
```
Comment (jonas-schievink) [inline client/shader/plant_dummy.tcs]: These should no longer exist.

File: client/shader/plant_dummy.tes [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    vec3 normal=(t1*t1*t1-3.0*t1*t1+3.0*t1)*tcs_normal*0.2;
+    /*vec3 x2=(t2*t2*t2-3.0*t2*t2+3.0*t2)*mid;
+    vec3 x3=(t3*t3*t3-3.0*t3*t3+3.0*t3)*cd;*/
+
+    tePosition = tePosition+normal;
+
+    gl_Position = proj_matrix * view_matrix * vec4(tePosition+offset, 1);
+
+    tes_color = tcPosition[0];
+}
```
Comment (jonas-schievink) [inline client/shader/plant_dummy.tes]: ^

File: client/src/game.rs [hunk 3]
```diff
@@ -11,11 +11,19 @@ use std::time::{Duration, Instant};
 use std::rc::Rc;
 use std::net::{SocketAddr, TcpStream};
 use std::error::Error;
+use base::world::World;
+use camera::Camera;
+use base::world::PillarSection;
+use base::world;
+// use base::world::HeightType; //WILL BE USED LATER
```
Comment (jonas-schievink) [inline client/src/game.rs]: Remove this

File: client/src/game.rs [hunk 4]
```diff
@@ -109,8 +131,97 @@ impl Game {
...
+    //                     pillar_section.remove(i);
+    //                 }
+    //             } else {
+    //                 return;
+    //             }
+    //         }
+    //         None => return,
+    //     };
+    //     self.world_manager.recalulate_chunk(Point3f::new(pos.x, pos.y, pos.z));
+    // }
```
Comment (jonas-schievink) [inline client/src/game.rs]: Remove commented code here too

File: client/src/world/chunk_std.frag [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+in vec3 x_color;
+in vec3 surfaceNormal;
+
+out vec3 color;
+
+const vec3 sun = normalize(vec3(1.0, 0.0, 1.0));
+void main() {
+    float diffuse = max(0.0, dot(sun, surfaceNormal));
+    color = x_color * 0.1 + x_color * diffuse;
+}
```
Comment (jonas-schievink) [inline client/src/world/chunk_std.frag]: Remove this, shaders don't belong here!

Comment (jonas-schievink) [inline client/src/world/chunk_std.frag]: Also why did this got duplicated?

File: client/src/world/world_view.rs [hunk 6]
```diff
@@ -88,6 +101,45 @@ impl WorldView {
...
+                    color: BlendingFunction::Addition {
+                        source: LinearBlendingFactor::SourceAlpha,
+                        destination: LinearBlendingFactor::OneMinusSourceAlpha,
+                    },
+                    ..Default::default()
+                },
+                ..Default::default()
+            };
+
+            // println!("DRAW: {:?}", self.outline.position().to_arr());
```
Comment (jonas-schievink) [inline client/src/world/world_view.rs]: Replace with uncommented `debug!` or remove

File: client/src/world_manager.rs [hunk 7]
```diff
@@ -165,6 +179,23 @@ impl WorldManager {
             }
         }
     }
+
+    // NOT USED UNTIL NOW
```
Comment (jonas-schievink) [inline client/src/world_manager.rs]: Until now?

File: client/src/world_manager.rs [hunk 8]
```diff
@@ -165,6 +179,23 @@ impl WorldManager {
...
+    pub fn recalulate_chunk(&mut self, pos: Point3f) {
```
Comment (jonas-schievink) [inline client/src/world_manager.rs]: Typo

File: client/src/world_manager.rs (hunk 7 above)
Comment (jonas-schievink) [inline client/src/world_manager.rs]: This doesn't look used, can you remove it or explain why you'll use it later (and what for)?

File: client/src/game.rs (hunk 3 above)
Comment (NilsAffing) [inline client/src/game.rs]: We need this later. At the moment travis is complaining about warnings.

File: client/src/game.rs (hunk 4 above)
Comment (NilsAffing) [inline client/src/game.rs]: ^

File: client/src/world_manager.rs (hunk 7 above)
Comment (NilsAffing) [inline client/src/world_manager.rs]: Its for updating the chunk at the given position (If some Hexagon is replaced/destroyed)

**Comment by jonas-schievink** [comment]
The removing part doesn't yet work. That will be done later.

**Pull request merged**